src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/TypeLoaderExports.cs

```diff
@@ -133,7 +142,11 @@ internal static unsafe IntPtr OpenInstanceMethodLookup(IntPtr openResolver, obje
         private static Entry LookupInCache(Entry[] cache, IntPtr context, IntPtr signature)
```

Comment:
We may consider using similar hashtable here as what we use for casting. Both solve the same problem.

cc @VSadov